src/python/pants/backend/python/rules/python_test_runner_test.py

```diff
@@ -0,0 +1,18 @@
+# Copyright 2019 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+import pytest
+
+from pants.backend.python.rules.python_test_runner import get_timeout_seconds_for_target
+
+
+@pytest.mark.parametrize(
+  "target_timeout,default_timeout,maximum_timeout,result",
+  [
+    (10, 1, 2, 2), # configured timeout greater than maximum --> maximum
+    (2, 1, 2, 2), # configured timeout OK --> configured timeout
+    (None, 1, 2, 1), # target has no configured timeout --> default
+    (None, 10, 2, 2), # target has no configured timeout and default is greater than maximum --> maximum
+  ])
```

Comment:
Here, I don't think the parameteration brings more benefits than costs. The cost is that it's a little difficult to follow all the magic numbers, hence the need for comments. Instead, we can embed that information in the code itself, e.g.

```python
def test_timeout_greater_than_maximum() -> None:
  assert get_timeout_seconds_for_target(test_target_timeout=10, default_timeout=1, timeout_maximum=2) == 2

def test_timeout_valid_target_timeout() -> None:
  assert get_timeout_seconds_for_target(test_target_timeout=2, default_timeout=1, timeout_maximum=3) == 2
```